test/e2e/openshift/openshift_test.go

```diff
@@ -85,6 +85,62 @@ var _ = Describe("Azure Container Cluster using the OpenShift Orchestrator", fun
 		}
 	})
 
+	It("should whitelist pods on compute nodes", func() {
+		namespaces := []string{
+			"openshift-infra",
+			"openshift-node",
+			"openshift-metrics",
+			"kube-",
+			"default",
+		}
+
+		whitelist := []string{
+			"sync-",
+			"ovs-",
+			"sdn-",
+		}
+		list, err := pod.GetAll("")
+		Expect(err).NotTo(HaveOccurred())
+
+		for _, pod := range list.Pods {
+			skip := false
+			// we have a whitelist of namespaces that we will skip
+			for _, pattern := range namespaces {
+				if strings.Contains(pod.Metadata.Namespace, pattern) {
+					skip = true
+					break
+				}
+			}
+			if skip {
+				skip = false
+				continue
+			}
+
+			found := false
+			if strings.Contains(pod.Spec.Nodename, "compute") {
+				for _, pattern := range whitelist {
+					if strings.Contains(pod.Metadata.Name, pattern) {
+						found = true
+						break
+					}
+				}
+				Expect(found).To(BeTrue())
+			}
+		}
+	})
+
+	// Verify all pods have initialized, ready, podscheduled are true
+	It("should have all pods in ready state", func() {
```

Comment:
have you tested this passes against a cluster today? does it catch the asb problem that we've currently got? if so, to get this in, you might need to exclude the asb pods from the test and add a `// TODO: remove this when issue #xyz is closed`

Reply:
Will run and confirm that we catch this case.